addons/predicates/tkr.go

```diff
@@ -18,3 +22,38 @@ func TKR(log logr.Logger) predicate.Funcs {
 		GenericFunc: func(e event.GenericEvent) bool { return true },
 	}
 }
+
+// ClusterHasLabel checks if the cluster has the given label
+func ClusterHasLabel(label string, logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return processIfClusterHasLabel(label, e.ObjectNew, logger.WithValues("predicate", "updateEvent"))
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return processIfClusterHasLabel(label, e.Object, logger.WithValues("predicate", "createEvent"))
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return processIfClusterHasLabel(label, e.Object, logger.WithValues("predicate", "deleteEvent"))
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return processIfClusterHasLabel(label, e.Object, logger.WithValues("predicate", "genericEvent"))
+		},
+	}
+}
+
+func processIfClusterHasLabel(label string, obj client.Object, logger logr.Logger) bool {
+	kind := obj.GetObjectKind().GroupVersionKind().Kind
+	cluster := &clusterapiv1beta1.Cluster{}
+
+	if kind == "" || kind != cluster.Kind {
+		return true
+	}
+
+	if _, ok := obj.GetLabels()[label]; ok {
```

Comment:
If `obj.Labels == nil`, this is going to panic. So, need to check for that first.